qiskit_aer/quantum_info/states/aer_state.py

```diff
@@ -147,6 +155,11 @@ def _initialize_with_ndarray(self, data, copy):
 
         self._last_qubit = num_of_qubits - 1
 
+    def __del__(self):
+        """"destructor. Call close() if not called."""
+        if self._state in (_STATE.ALLOCATED, _STATE.MOVED):
+            self.close()
```

Comment:
`__del__` in Python isn't guaranteed to be called at all, even when the Python interpreter exits.  It _usually_ gets called, but it's not for sure.  A reliable way to ensure the finaliser _always_ gets called is [`weakref.finalize`](https://docs.python.org/3/library/weakref.html#weakref.finalize) - you'd call that immediately after allocating the memory in C++ space, I imagine.

Reply:
To call `self.close()` in a callback, `weakref.finalize()` needs to take `self` as an argument for the callback. However, taking `self` means `self` will never be collected. I guess that another way is necessary.

However, `AerState` is a private class. `AerStatevector` will not keep `AerState` as `_STATE.ALLOCATED`. I believe that `__del__` is not necessary.